ci/build-package-set.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+shopt -s nullglob
+
+psroot=$(dirname "$(dirname "$(realpath "$0")")")
+
+if [[ "${CI:-}" && "$(echo $psroot/CHANGELOG.d/breaking_*)" ]]; then
+  echo "Skipping package-set build due to unreleased breaking changes"
+  exit 0
+fi
+
+tmpdir=$(mktemp -d)
+trap 'rm -rf "$tmpdir"' EXIT
+export PATH="$tmpdir/node_modules/.bin:$PATH"
+cd "$tmpdir"
+
+echo ::group::Ensure Spago is available
+which spago || npm install spago
+echo ::endgroup::
+
+echo ::group::Create dummy project
+echo 'let upstream = https://github.com/purescript/package-sets/releases/download/XXX/packages.dhall in upstream' > packages.dhall
+echo '{ name = "my-project", dependencies = [] : List Text, packages = ./packages.dhall, sources = [] : List Text }' > spago.dhall
+spago upgrade-set
```

Comment:
Perhaps passing in the `-P` / `--no-psa` global flag here might fix things? That's one thing I never tried in my PR.